src/addon-manager-proxy.ts

```diff
@@ -610,9 +695,12 @@ class AddonManagerProxy extends EventEmitter {
    * @method sendPairingPrompt
    * Send a prompt to the UI notifying the user to take some action.
    */
-  sendPairingPrompt(adapter, prompt, url = null, device = null) {
-    const data = {
-      adapterId: adapter.id,
+  sendPairingPrompt(adapter: Adapter, prompt: string, url?: string,
+                    device?: Device): void {
+    const data: MessageData8 = {
```

Comment:
Question: Where do these numbered types come from, e.g. `MessageData8`, `MessageData14`, `Message2`, etc.?

Reply:
They are generated by `json-schema-to-typescript`.
Because all data properties have the title `Message Data` it appends a number to the name so it becomes unique.
I can change the titles in the schema repository if you want.

Reply:
Ahh, I see. Changing the titles would certainly make this more understandable and easier to maintain.

Reply:
Done